python/afdko/pdflib/fontpdf.py

```diff
@@ -1989,7 +1989,12 @@ def makeWaterfallPDF(params, pdfFont, doProgressBar):
 
 	numWaterfallsOnPage = int(pageHeight / waterfallHeight)
 	numWaterFalls = len(glyphLists)
-	numPages = int(ceil(float(numWaterFalls) / numWaterfallsOnPage))
+
+	if numWaterfallsOnPage > 0:
+		numPages = int(ceil(float(numWaterFalls) / numWaterfallsOnPage))
+	else:
+		numPages = int(ceil(float(numWaterFalls)*pageHeight / waterfallHeight))
```

Comment:
I think the code at line 1996 should be:
```c
numPages = int(ceil(float(numWaterFalls) * waterfallHeight / pageHeight))
```
to be more like the calculation in line 1994.

Alternatively, perhaps you could replace lines 1993-1996 with:
```c
if numWaterfallsOnPage > 0:
	numWaterfallsOnPage = 1
numPages = int(ceil(float(numWaterFalls) / numWaterfallsOnPage))
```
similar to what is done in lines 1998-1999.

Reply:
I tried both of these suggestions locally using the example font and command from #1092 , and the only difference that I'm able to discern is the total page count displayed in the header.

In my original commit (using code suggested by @readroberts ), the total page count is shown as 16 (which is incorrect; the actual number of pages is 28).

Using the first method suggested above (change line 1996), the total page count is shown as 26, which is closer, but still wrong. For the alternative method above, the page count is 20 (NOTE: I tried to attach the PDF results of both of these, but GitHub is not letting me right now).

I want to be clear here that the _actual number of pages generated_ for all three of these is 28, and the waterfall content on the pages is identical, apart from the "## of ##" page indicator and timestamp in the headers.